### PR TITLE
ramips: add Asus RP-AC56 support

### PR DESCRIPTION
**Pull request by amdodak**
This change adds support for Asus RP-AC56 which
    is a dual band dual core 802.11ac access point

    CPU:   MT7621A dual-core 880MHz
    RAM:   64MB DDR2
    FLASH: 16MB NOR SPI
    WIFI:  2.4GHz 2x2 MT7603 b/g/n PCI
    WIFI:  5GHz 2x2 MT7662 a/b/ac PCI
    ETH:   1xLAN 1000base-T integrated
    LED:   Power, 2.4GHz WIFI, 5GHz WIFI
    BTN:   WPS, Reset
    UART:  Near ETH port, from ETH: 3V3-TxD-GND-RxD 57600 8n1
    MISC:  Audio support

    During the build, a factory update image is generated. This
    image can be used to upgrade from stock firmware.
    Another option to re-flash the device would be to hold "reset"
    button while powering it up and wait for power LED to turn
    orange. After that, an image: stock firmware or OpenWRT
    sysupdate can be pushed to 192.168.1.1 using TFTP protocol
    in binary mode

### CODE REVIEW

File: target/linux/ramips/base-files/lib/ramips.sh [hunk 1]
```diff
@@ -424,6 +424,9 @@ ramips_board_detect() {
 	*"RN502J")
 		name="xdxrn502j"
 		;;
+	*"RP-AC56")
+		name="rp-ac56"
+		;;
```
Comment (adrianschmutzler) [inline target/linux/ramips/base-files/lib/ramips.sh]: Remove this.

File: target/linux/ramips/image/mt7621.mk [hunk 2]
```diff
@@ -406,6 +410,17 @@ define Device/re6500
...
+define Device/rp-ac56
+  DTS := RP-AC56
+  BLOCKSIZE := 64k
+  IMAGE_SIZE := $(ralink_default_fw_size_16M)
+  DEVICE_TITLE := Asus RP-AC56
+  DEVICE_PACKAGES := kmod-mt7603 kmod-mt76x2 wpad kmod-sound-mt7620
+  IMAGES += factory.trx
+  IMAGE/factory.trx := append-kernel | append-rootfs | pad-rootfs | append-metadata | mkrtn56uimg -p RP-AC56 -f | check-size $$$$(IMAGE_SIZE)
+endef
+TARGET_DEVICES += rp-ac56
```
Comment (adrianschmutzler) [inline target/linux/ramips/image/mt7621.mk]: Please use `asus_rp-ac56` here

File: target/linux/ramips/image/mt7621.mk [hunk 3]
```diff
@@ -406,6 +410,17 @@ define Device/re6500
 endef
 TARGET_DEVICES += re6500
 
+define Device/rp-ac56
```
Comment (adrianschmutzler) [inline target/linux/ramips/image/mt7621.mk]: Please use `asus_rp-ac56` here

File: target/linux/ramips/base-files/lib/preinit/04_handle_checksumming [hunk 4]
```diff
@@ -35,8 +35,8 @@ do_checksumming_disable() {
 			echo "Checksum is already zero, nothing to do."
 		fi
 	;;
-	rt-n56u)
-		echo "Board is ASUS RT-N56U, replacing uImage header..."
+	rp-ac56|rt-n56u)
```
Comment (adrianschmutzler) [inline target/linux/ramips/base-files/lib/preinit/04_handle_checksumming]: Please use `asus,rp-ac56` here

File: target/linux/ramips/base-files/etc/board.d/02_network [hunk 5]
```diff
@@ -376,6 +376,9 @@ ramips_setup_interfaces()
 		ucidef_add_switch "switch0" \
 			"0:lan:1" "1:lan:2" "2:lan:3" "3:lan:4" "6@eth0"
 		;;
+	rp-ac56)
```
Comment (adrianschmutzler) [inline target/linux/ramips/base-files/etc/board.d/02_network]: Please use `asus_rp-ac56` here

File: target/linux/ramips/base-files/etc/board.d/01_leds [hunk 6]
```diff
@@ -299,6 +299,10 @@ re350-v1)
 	ucidef_set_led_netdev "eth_act" "LAN act" "$boardname:green:eth_act" "eth0" "tx rx"
 	ucidef_set_led_switch "eth_link" "LAN link" "$boardname:green:eth_link" "switch0" "0x01"
 	;;
+rp-ac56)
```
Comment (adrianschmutzler) [inline target/linux/ramips/base-files/etc/board.d/01_leds]: Please use `asus_rp-ac56` here

**Comment by adrianschmutzler** [comment]
There has been a reorganization in ramips target today. Please:
- update your board.d subfiles since they have been re-sorted
- update your Makefile (mt****/rt****.mk) changes so sorting of devices is correct
- rename your DTS/DTSI files following the scheme soc_vendor_device.dts(i), e.g. mt7620a_tplink_archer-c50-v1.dts (and update includes there if necessary)
- remove `DTS := ...` from Makefile
- add `MTK_SOC` to Makefile (needs to match "soc" in DTS file name), e.g. `MTK_SOC := mt7620a`

Thank you!

**Comment by ynezz** [comment]
Ping.

**Comment by adrianschmutzler** [comment]
This has not seen any response from the author after the initial submission. I think it can be closed.

**Pull request closed without merging**